File: frameworks/node/koa/UnitTests/template.test.ts
import assert from "node:assert/strict";
import { test } from "node:test";

import { client, expected, normal } from "./app.ts";

// rb:test template.small,template.medium
for (const [id, size] of [["template.small", "small"], ["template.medium", "medium"]] as const) {
  test(`${id}: the payload is rendered by the EJS template`, async () => {
    const response = await client.get(`/template/${size}`);

    assert.equal(response.status, 200);
    assert.match(response.headers["content-type"]!, /^text\/html/);
    assert.equal(normal(response.text), expected.page(`items.${size}.json`));
  });
}

File: frameworks/go/huma/UnitTests/template_test.go
package unittests

import (
	"net/http"
	"testing"
)

// rb:test template.small,template.medium
func TestThePageIsRendered(t *testing.T) {
	for _, row := range []struct{ id, path, file string }{
		{"template.small", "/template/small", "items.small.json"},
		{"template.medium", "/template/medium", "items.medium.json"},
	} {
		t.Run(row.id, func(t *testing.T) {
			response := client(t).Get(row.path)

			assertStatus(t, response, http.StatusOK)
			assertHeader(t, response, "Content-Type", "text/html; charset=utf-8")
			if got, want := normal(response.Body.String()), normal(page(t, row.file)); got != want {
				t.Fatalf("expected %s\n     got %s", want, got)
			}
		})
	}
}
